Mathlib/Geometry/Manifold/IsManifold/Basic.lean

```diff
@@ -252,12 +275,60 @@ theorem target_eq : I.target = range (I : H → E) := by
   rw [← image_univ, ← I.source_eq]
   exact I.image_source_eq_target.symm
 
-protected theorem uniqueDiffOn : UniqueDiffOn 𝕜 (range I) :=
-  I.target_eq ▸ I.uniqueDiffOn'
+theorem nonempty_interior : (interior (range I)).Nonempty :=
+  I.nonempty_interior'
+
+theorem range_eq_univ_of_not_isRCLikeNormedField (h : ¬ IsRCLikeNormedField 𝕜) :
+    range I = univ := by
+  simpa [h] using I.convex_range'
+
+/-- If a set is `ℝ`-convex for some normed space structure, then it is `ℝ`-convex for the
+normed space structure coming from an `IsRCLikeNormedField 𝕜`. Useful when constructing model
+spaces to avoid diamond issues when populating the field `convex_range'`. -/
+lemma _root_.Convex.convex_isRCLikeNormedField [NormedSpace ℝ E] [h : IsRCLikeNormedField 𝕜]
```

Comment:
I think it would be valuable to have a constructor for `ModelWithCorners` for Real-valued manifolds that already uses this lemma to simplify what to prove for the `convex_range'` field (similar to `ModelWithCorners.of_target_univ`).

Reply:
I have added the constructor. Applying it is more complicated than applying the lemma `Convex.convex_isRCLikeNormedField`, because one needs first to construct the partial equiv separately, though, so in the concrete constructions I've still kept the old version.